### PR TITLE
Demonstrate annotating structured metadata to classification functions

### PR DESCRIPTION
**Pull request by cthoyt**
# Summary
 
This PR demonstrates a potential solution to #28 in a pilot applied to classification metrics. It could be extended in a future pull request to other branches of the package.
 
- [x] Code passes all tests
- [x] Unit tests provided for these changes
- [x] Documentation and docstrings added for these changes

## Changes 

- [x] Add an `annotate` decorator that adds some structured information to classification functions
- [x] Add tests to make sure the data is accessible
- [x] Add unit test to ensure all classification functions are annotated (1afd68b)
- [x] Annotate all classification functions

## Future

Before finalizing this PR, I had also used the `annotate` function to make a registry of functions. This could be used to make the generation/maintenance of the ClassificationMetricSet much easier, but I'd save that for a different PR.

### CODE REVIEW

File: tests/unit/test_metrics.py [hunk 1]
```diff
@@ -71,6 +73,25 @@ def setUp(self):
         self.y_true = np.array([0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0])
         self.y_score = np.array([1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 0])
 
+    @unittest.skip("Not ready to commit to requiring this in PR https://github.com/AstraZeneca/rexmex/pull/29")
```
Comment (cthoyt) [inline tests/unit/test_metrics.py:76]: remove this if/when interested in committing to annotating all classification functions

File: rexmex/metrics/classification.py [hunk 2]
```diff
@@ -339,13 +451,23 @@ def prevalence_threshold(y_true: np.array, y_score: np.array) -> float:
...
+        - https://en.wikipedia.org/wiki/Sensitivity_and_specificity#Prevalence_threshold
+        - https://dx.doi.org/10.1371%2Fjournal.pone.0240215
     """
     fpr = false_positive_rate(y_true, y_score)
     tpr = true_positive_rate(y_true, y_score)
     pthr = (fpr ** 0.5) / (fpr ** 0.5 + tpr ** 0.5)
     return pthr
 
 
+@classifications.annotate(  # TODO
```
Comment (cthoyt) [inline rexmex/metrics/classification.py:466]: help wanted - I'm not really sure what the bounds for this are nor whether bigger is better

File: rexmex/metrics/classification.py [hunk 3]
```diff
@@ -363,6 +485,11 @@ def threat_score(y_true: np.array, y_score: np.array) -> float:
     return ts
 
 
+@classifications.annotate(  # TODO
```
Comment (cthoyt) [inline rexmex/metrics/classification.py:488]: help wanted - I'm not really sure what the bounds for this are nor whether bigger is better

**Comment by benedekrozemberczki** [comment]
Thank you @cthoyt!

**Pull request merged**